File: Backend/src/main/java/dev/hv/model/enums/UserPermissions.java
package dev.hv.model.enums;

public enum UserPermissions
{
    READ,
    WRITE,
    UPDATE,
    DELETE;

    public static UserPermissions translateHttpToUserPermission(String methodeType){
        return switch (methodeType)
        {
            case "POST" -> UserPermissions.WRITE;
            case "GET" -> UserPermissions.READ;
            case "PUT" -> UserPermissions.UPDATE;
            case "DELETE" -> UserPermissions.DELETE;
            case null, default -> null;
        };
    }
}

File: Frontend/P_3FA132_Gruppe_3_Frontend/Data/Models/Enums/UserPermissions.cs
namespace P_3FA132_Gruppe_3_Frontend.Data.Models.Enums
{
    public enum UserPermissions
    {
        READ, 
        WRITE, 
        UPDATE, 
        DELETE
    }

    public static class UserPermissionsExtensions
    {
        public static UserPermissions ToUserPermissions(this string userPermissionsString)
        {
            return userPermissionsString.ToUpper() switch
            {
                "READ" => UserPermissions.READ,
                "WRITE" => UserPermissions.WRITE,
                "UPDATE" => UserPermissions.UPDATE,
                "DELETE" => UserPermissions.DELETE,
                _ => throw new Exception("UserPermissions not found")
            };
        }
    }
}
